### DISCUSSION TITLE
OAuth2 Sign Up with Admin-only creation rule

### DISCUSSION THREAD

**Discussion by ElliotHYLee** [pocketbase/pocketbase#2639] [Q&A]
In the collection of "users", the creation rule is admin only. I like this. As I have a helper server to hanlde creation using the admin api.

When a new user tries to sign up, say with Google, the PB's api returns 400 or so. Good. Yet, is it possible to get the user's email at least? I guess the PB's backend somehow knows the user email as the google sign in with the redirect url handles it.

Once my front-end knows the email at least, the helper backend can create the user with the email. At the next attempt, the user can login .

In fact, is it possible to see the google's redirect information such as user email, first & last name by configuring the PB's hook? Regardless of user login/creation, once the SSO returns the good result, it would be much helpful to see the names as well assuming the scope is set.

**Comment by ganigeorgiev** [comment]
You can try attaching to the [`OnRecordBeforeAuthWithOAuth2Request` hook](https://pocketbase.io/docs/event-hooks/#record-auth-api-hooks). It should be triggered after the OAuth2 token exchange giving you access to the OAuth2 user, but before the new PocketBase user creation (_you can find all available hook event fields in [`core.RecordAuthWithOAuth2Event`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.16.4/core#RecordAuthWithOAuth2Event)_).
